plugins/by-name/neorg/default.nix

```diff
@@ -1,173 +1,222 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
 let
-  cfg = config.plugins.neorg;
+  inherit (lib.nixvim)
+    defaultNullOpts
+    mkNullOrStr
+    mkNullOrOption
+    mkNullOrOption'
+    ;
+  inherit (lib) types;
 in
-with lib;
-{
-  options.plugins.neorg = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "neorg";
-
-    package = lib.mkPackageOption pkgs "neorg" {
-      default = [
-        "vimPlugins"
-        "neorg"
-      ];
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "neorg";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # TODO introduced 2024-12-12: remove after 25.05
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "lazyLoading"
+    [
+      "logger"
+      "plugin"
+    ]
+    [
+      "logger"
+      "useConsole"
+    ]
+    [
+      "logger"
+      "highlights"
+    ]
+    [
+      "logger"
+      "useFile"
+    ]
+    [
+      "logger"
+      "level"
+    ]
+    [
+      "logger"
+      "floatPrecision"
+    ]
+  ];
+  imports = [
+    ./deprecations.nix
+  ];
+
+  extraOptions = {
+    telescopeIntegration = {
+      enable = lib.mkEnableOption "`neorg-telescope` plugin for telescope integration.";
+
+      package = lib.mkPackageOption pkgs "neorg-telescope" {
+        default = [
+          "vimPlugins"
+          "neorg-telescope"
+        ];
```

Comment:
```suggestion
        ];
        nullable = true;
```

Reply:
I don't really want it nullable...
Well actually maybe it is stupid to have `enable` + non-`null`able-`package`.
Maybe we should only have a `null`able `neorg-telescopePackage` ?

Reply:
Oh, is the only thing we're doing in here installing the package? Normally we have to add the module name to our telescope module

Reply:
Here it's a plugin install indeed.

Reply:
If it doesn't require `require('telescope').load_extension(extension)` and we're just adding the package, then I agree it doesn't make sense to be nullable. It also feels too small to be its own module.

Reply:
Ok, then I would say we keep the non-`null`able `package` + `enable` options.
Indeed, if a user wants to enable the feature, he only has to `telescopeIntegration.enable = true;` and not have to manually set a plugin value.